pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -60,34 +60,45 @@ func (v *ControlPlaneMutator) Handle(ctx context.Context, req admission.Request)
 	// on create we set the version to the current default version
 	// on update, if the version is removed we reset it to what was previously set
 	currentVersion := mutator.NewVersion()
-	effectiveVersion, _ := versions.ParseVersion(currentVersion)
 	if currentVersion == "" {
 		switch req.AdmissionRequest.Operation {
 		case admissionv1beta1.Create:
 			log.Info("Setting .spec.version to default value", "version", versions.DefaultVersion.String())
-			mutator.SetVersion(mutator.DefaultVersion())
-			effectiveVersion, _ = versions.ParseVersion(mutator.DefaultVersion())
+			currentVersion = mutator.DefaultVersion()
 		case admissionv1beta1.Update:
 			oldVersion := mutator.OldVersion()
 			if currentVersion != oldVersion && oldVersion != versions.InvalidVersion.String() {
 				log.Info("Setting .spec.version to existing value", "version", oldVersion)
-				mutator.SetVersion(oldVersion)
-				effectiveVersion, _ = versions.ParseVersion(oldVersion)
+				currentVersion = oldVersion
 			}
 		}
 	}
+	mutator.SetVersion(currentVersion)
+	effectiveVersion, _ := versions.ParseVersion(currentVersion)
 
-	if req.AdmissionRequest.Operation == admissionv1beta1.Create {
-		// As we are deprecating IOR, on creating a v2.5 SMCP we want to disable IOR if not specified explicitly
-		newOpenShiftRoute := mutator.IsOpenShiftRouteEnabled()
-
-		if newOpenShiftRoute == nil && effectiveVersion.AtLeast(versions.V2_5.Version()) {
-			mutator.SetOpenShiftRouteEnabled(false)
-		}
+	if req.AdmissionRequest.Operation == admissionv1beta1.Update {
+		oldsmcp := &v2.ServiceMeshControlPlane{}
+		err := v.decoder.DecodeRaw(req.AdmissionRequest.OldObject, oldsmcp)
```

Comment:
This assumes that the webhook is mutating a v2 SMCP, which isn't always the case. I might be wrong, but it looks like this is why we have the `smcpmutator` (so that the we can have two implementations - for v1 and for v2).